mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -276,11 +276,13 @@ private[spark] object BLAS extends Serializable with Logging {
 
     var i = 0
     while (i < nnz) {
-      val multiplier = alpha * xValues(i)
-      val offset = xIndices(i) * mA
-      var j = 0
+      var j = i
```

Comment:
If you start at `i+1`, you don't even need to check if `i != j`. Is this notably faster at reasonable scale?